evalml/model_understanding/feature_explanations.py

```diff
@@ -0,0 +1,208 @@
+                "time series binary": "log loss binary",
+                "time series multiclass": "log loss multiclass",
+            }[pipeline.problem_type.value]
+
+        if X is None or y is None:
+            raise ValueError(
+                "X and y are required parameters for explaining pipelines with permutation importance."
+            )
+
+        X = infer_feature_types(X)
+        y = infer_feature_types(y)
+        imp_df = calculate_permutation_importance(pipeline, X, y, objective)
+    elif importance_method == "feature":
+        objective = None
+        imp_df = pipeline.feature_importance
+    else:
+        raise ValueError(f"Unknown importance method {importance_method}")
+
+    linear_importance = False
+    if (
+        pipeline.estimator.model_family == ModelFamily.LINEAR_MODEL
+        and importance_method == "feature"
+    ):
+        linear_importance = True
+
+    (
+        most_important_features,
+        somewhat_important_features,
+        detrimental_features,
+    ) = get_influential_features(
+        imp_df,
+        max_features,
+        min_importance_threshold,
+        linear_importance,
+    )
+    target = y if y is None else y.name
+    explanation = _fill_template(
+        pipeline.estimator,
+        target,
+        objective,
+        most_important_features,
+        somewhat_important_features,
+        detrimental_features,
+    )
+    logger.info(explanation)
+
+
+def get_influential_features(
+    imp_df, max_features=5, min_importance_threshold=0.05, linear_importance=False
+):
+    """Finds the most influential and detrimental features from a dataframe of feature importances.
+
+    Args:
+        imp_df (pd.DataFrame): DataFrame containing feature names and associated importances.
+        max_features (int): The maximum number of features to include in an explanation. Defaults to 5.
+        min_importance_threshold (float): The minimum percent of total importance a single feature can have to be considered important. Defaults to 0.05.
+        linear_importance (bool): When True, negative feature importances are not considered detrimental. Defaults to False.
+
+    Returns:
+        (list, list, list): Lists of feature names corresponding to heavily influential, somewhat influential, and detrimental features, respectively.
+
+    """
+    heavy_importance_threshold = max(0.2, min_importance_threshold + 0.1)
+
+    # Separate negative and positive features, if situation calls
+    if linear_importance:
+        pos_imp_df = imp_df
+        pos_imp_df["importance"] = abs(pos_imp_df["importance"])
+        neg_imp_df = pos_imp_df[pos_imp_df["importance"] < 0]
+    else:
+        neg_imp_df = imp_df[imp_df["importance"] < 0]
+        pos_imp_df = imp_df[imp_df["importance"] >= 0]
+
+    # Normalize the positive features to sum to 1
+    pos_imp_df["importance"] = pos_imp_df["importance"] / sum(pos_imp_df["importance"])
+
+    num_feats = min(len(pos_imp_df), max_features)
```

Comment:
Based on the examples, this doesn't limit the number of features identified as detracting from the model performance, right? From https://feature-labs-inc-evalml--2861.com.readthedocs.build/en/2861/user_guide/model_understanding.html#Human-Readable-Importance:


"Random Forest Classifier: The log loss binary performance is heavily influenced by worst perimeter, and is somewhat influenced by worst area, worst radius, mean concave points, and worst concave points.
The features smoothness error, mean smoothness, fractal dimension error, worst compactness, mean concavity, mean compactness, radius error, mean radius, worst fractal dimension, symmetry error, perimeter error, concave points error, and worst symmetry detracted from model performance. We suggest removing these features."